src/styles/main.scss

```diff
@@ -1,3 +1,58 @@
+@import "./variables";
+
 body {
   margin: 0;
+  display: flex;
+  flex-direction: column;
+  height: 100vh;
+  justify-content: center;
+}
+
+.calendar {
+  display: flex;
+  width: $column-width;
+  flex-wrap: wrap;
+  margin: 0 auto;
+  gap: $gap;
+
+  @for $month-length from 28 through 31 {
+    &--#{$month-length} :nth-child(#{$month-length}) {
+      & ~ .calendar__day {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    width: $day-width;
+    height: $day-height;
+    background-color: $main-color;
+    box-sizing: border-box;
+    border: 1px solid black;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+      transition-duration: 0.5s;
```

Comment:
The transition property should be applied to the element, not on hover. If you apply it on hover, the transition will only work when you hover over the element, but not when you stop hovering.